server/platform/services/sharedchannel/sync_send_remote.go

```diff
@@ -691,6 +703,203 @@ func (scs *Service) sendProfileImageSyncData(sd *syncData) {
+		users: make(map[string]*model.User),
+	}
+
+	// Collect users to sync
+	users, latestTimestamp, _, hasMore, err := scs.collectUsersForGlobalSync(rc, batchSize)
+	if err != nil {
+		return err
+	}
+
+	// Exit early if no users to sync
+	if len(users) == 0 {
+		scs.server.Log().Log(mlog.LvlSharedChannelServiceDebug, "No users to sync for remote cluster",
+			mlog.String("remote_id", rc.RemoteId))
+		return nil
+	}
+
+	// Add users to sync data
+	sd.users = users
+
+	// Send the collected users to remote
+	if err := scs.sendUserSyncData(sd, latestTimestamp); err != nil {
+		scs.server.Log().Log(mlog.LvlSharedChannelServiceError, "Error sending user batch during sync",
+			mlog.String("remote_id", rc.RemoteId),
+			mlog.Err(err),
+		)
+		return fmt.Errorf("error sending user batch during sync: %w", err)
+	}
+
+	// Schedule next batch if needed
+	if hasMore {
+		scs.scheduleNextUserSyncBatch(rc, latestTimestamp, batchSize, len(users))
+	}
+
+	return nil
+}
+
+// getGlobalUserSyncBatchSize returns the configured batch size for user syncing
+func (scs *Service) getGlobalUserSyncBatchSize() int {
+	batchSize := MaxUsersPerSync
+	if scs.server.Config().ConnectedWorkspacesSettings.GlobalUserSyncBatchSize != nil {
+		configValue := *scs.server.Config().ConnectedWorkspacesSettings.GlobalUserSyncBatchSize
+		if configValue > 0 && configValue <= 200 {
+			batchSize = configValue
+		}
+	}
+	return batchSize
+}
+
+// collectUsersForGlobalSync fetches users that need to be synced to the remote
+func (scs *Service) collectUsersForGlobalSync(rc *model.RemoteCluster, batchSize int) (map[string]*model.User, int64, int, bool, error) {
+	options := &model.UserGetOptions{
+		Page:    0,
+		PerPage: 100, // Database fetch batch size
+		Active:  true,
+		Sort:    "update_at_asc", // Order by UpdateAt ASC to ensure cursor consistency
+	}
+
+	// Only use UpdatedAfter for incremental syncs, not the initial sync
+	// This ensures users with UpdateAt=0 are included in the first sync
+	if rc.LastGlobalUserSyncAt > 0 {
+		options.UpdatedAfter = rc.LastGlobalUserSyncAt
+	}
+
+	users := make(map[string]*model.User)
+	latestTimestamp := rc.LastGlobalUserSyncAt
+	totalCount := 0
+
+	// Page through database results
+	for {
+		batch, err := scs.server.GetStore().User().GetAllProfiles(options)
+		if err != nil {
+			scs.server.Log().Log(mlog.LvlSharedChannelServiceError, "Error fetching users for global sync",
+				mlog.String("remote_id", rc.RemoteId),
+				mlog.Err(err),
+			)
+			return nil, 0, 0, false, err
+		}
+
+		if len(batch) == 0 {
+			break // No more users to process
+		}
+
+		totalCount += len(batch)
+
+		// Process each user in this database page
+		for _, user := range batch {
+			// Stop if we've reached batch limit
+			if len(users) >= batchSize {
+				return users, latestTimestamp, totalCount, true, nil
+			}
+
+			// Skip users from target remote to avoid syncing them back
+			if user.IsRemote() && user.GetRemoteID() == rc.RemoteId {
+				continue
+			}
```

Comment:
I think we should only sync local users, to avoid remote users being automatically synced with a server that they don't share any channel with:

```suggestion
			// Skip remote users as the goal is to make all our local users discoverable
			// after the initial connection
			if user.IsRemote() {
				continue
			}
```